x-pack/plugin/repository-encrypted/src/main/java/org/elasticsearch/repositories/encrypted/EncryptedRepository.java

```diff
@@ -0,0 +1,16 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.repositories.encrypted;
+
+public class EncryptedRepository {
+    static final int GCM_TAG_LENGTH_IN_BYTES = 16;
+    static final int GCM_IV_LENGTH_IN_BYTES = 12;
+    static final int AES_BLOCK_SIZE_IN_BYTES = 128;
+    static final String GCM_ENCRYPTION_SCHEME = "AES/GCM/NoPadding";
+    static final long PACKET_START_COUNTER = Long.MIN_VALUE;
```

Comment:
Is there a reason not to just start at `0` ? Do we really need a full 64bits of counter space?

Reply:
No, there shouldn't be a practical need for the full range of a 64bit counter. But I kinda like the cleanliness of the wrap-around condition in encryption where you start off with the smallest value for the counter, which you increment and check that after increment it's not equal to the start value. Negative values might give the impression that something must be wrong, but when looking at the byte sequence of the IV the leading "1" of the two's complement representation is not so confusing (there is still a block of 31 zeroes that are incremented in the usual way).

I'll keep the current approach, but If you otherwise prefer we define a min (0) and max (long.max) value for the counter, I'm OK with this too.